lessons/programming/algorithm/index.mdx

```diff
@@ -8,14 +8,69 @@ title: Algorithm
+
+An algorithm can be written by a plain langugae, a pseudo-code or described by a diagram-representd flowchart and it can be implemented in different computer programming langugages. 
+
+## Pseudo-code
+Pseudo-codes are simplified version of computer programs and written in nantural language.
+
+- Pseudo-code can be read and followed by programmers to write computer program.
+- Pseudo-code does not have specific syntax and can't be excuted by a computer.
+- Pseudo-code can be implemented in specific programming language.
+
+for example, a pseudo-code of counting members is as follows:
+
+num_members_far is set to be zero before counting
+while the input of member name is not empty
+    increment number_members_so_far by one
+num_members_so_far is the total number of members at the end
 
 ## Flowchart
 An algorithm can be described as a pictorial representation with different symbols, shapes, arrow that describes the different process. Each shape describes a different process:
 
 - The process is represented graphically.
-- The process is described block by block.
+- The process is described block by block with the arrows between blocks indicating the order of workflow.
 - The described steps are easy to follow.
 - Cost less time and does not require the understanding of code.
+ 
+for example, a flowchart of counting members is as follows:
+        --------------------------------
+        |    num_members_so_far = 0     |
+        ---------------------------------
+                      |
+                      V
+                      *
+                    ******
+                  **********                      otherwise
+     ------->* while input name is not empty *-------------
+     |            **********                              |
+     |              *******                               |
+     |                 *                                  |
+     |                 |                                  |
+     |                 V                                  |
+     |    -------------------------------------------     |
+     ----|   increment number_members_so_far by one |     |
+         --------------------------------------------     |
+                                                          |
+                                                          |
+                                                          V
+         ---------------------------------------------------------
+         |  output number_members_so_far as total_number_members |
+         ----------------------------------------------------------
+
+## Example of Python code implementation
+For reference, the above pseudo-code or flowchart can be followed to write a Python code as follows:
+
+number_members_so_far = 0
+while (input("name? ") != ""):
+     number_members_so_far += 1
+print(number_members_so_far)
```

Comment:
You can use the Python markdown syntax highlighting tag:
```
```python
code
end with ```
```

```python
number_members_so_far = 0
while (input("name? ") != ""):
     number_members_so_far += 1
print(number_members_so_far)
```